### PR TITLE
WarpRun improvements

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

File: warp10/src/main/sh/warp10-standalone.sh [hunk 1]
```diff
@@ -569,6 +569,11 @@ tokengen() {
   ${JAVACMD} -cp ${WARP10_JAR} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 io.warp10.worf.TokenGen ${CONFIG_FILES} $2 -
 }
 
+run() {
+  getConfigFiles
+  ${JAVACMD} -cp ${WARP10_JAR} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 -Dwarp10.config="${CONFIG_FILES}" io.warp10.WarpRun $1
```
Comment (steven-gueguen) [inline warp10/src/main/sh/warp10-standalone.sh:574]: Should be $2 -

**Review by steven-gueguen — COMMENTED** [review]
Doc is missing for new function

**Review by steven-gueguen — APPROVED** [review]

**Comment by hbs** [comment]
PTAL, the calling parameters were incorrect for WarpRun, the extraneous '-' needed to be removed.

File: warp10/src/main/java/io/warp10/WarpRun.java [hunk 2]
```diff
@@ -25,61 +25,97 @@
...
+      // Register the WarpRun and Debug extensions
+      //
+
+      WarpScriptLib.register(new WarpRunWarpScriptExtension());
+      WarpScriptLib.register(new DebugWarpScriptExtension());
+
       MemoryWarpScriptStack stack = new MemoryWarpScriptStack(null, null, properties);
       stack.maxLimits();
-      
+
```
Comment (steven-gueguen) [inline warp10/src/main/java/io/warp10/WarpRun.java:80]: Break condition should be change at line 94

**Review by steven-gueguen — CHANGES_REQUESTED** [review]

File: warp10/src/main/java/io/warp10/WarpRun.java (hunk 2 above)
Comment (hbs) [reply warp10/src/main/java/io/warp10/WarpRun.java:80]: ?

Comment (steven-gueguen) [reply warp10/src/main/java/io/warp10/WarpRun.java:81]: When dealing with a "-" parameter, the reading process never end.
The `if (null == line) {` should be adapted.

Comment (hbs) [reply warp10/src/main/java/io/warp10/WarpRun.java:81]: When stdin reaches EOF, readline will return null, do you have a test case where that is not the case?

Comment (steven-gueguen) [reply warp10/src/main/java/io/warp10/WarpRun.java:81]: No, I misused it

**Review by steven-gueguen — APPROVED** [review]

**Pull request merged**